drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-client/src/main/java/org/drools/workbench/screens/scenariosimulation/kogito/client/editor/ScenarioSimulationEditorKogitoWrapper.java

```diff
@@ -212,6 +220,38 @@ public void validate(Simulation simulation, Settings settings, RemoteCallback<?>
         //
     }
 
+    @Override
+    public void onRefreshedModelContent(SimulationRunResult testResultMessage) {
+        /* No actions required after data refresh in Kogito */
+    }
+
+    @Override
+    public void populateDocks(String identifier) {
+        switch (identifier) {
+            case SettingsPresenter.IDENTIFIER:
+                scenarioSimulationKogitoDocksHandler.getSettingsPresenter().ifPresent(presenter -> {
+                    scenarioSimulationEditorPresenter.setSettings(presenter);
+                    presenter.setCurrentPath(scenarioSimulationEditorPresenter.getPath());
+                });
+                break;
+            case TestToolsPresenter.IDENTIFIER:
+                scenarioSimulationKogitoDocksHandler.getTestToolsPresenter().ifPresent(
+                        presenter -> scenarioSimulationEditorPresenter.setTestTools(presenter));
+                break;
+            case CheatSheetPresenter.IDENTIFIER:
+                scenarioSimulationKogitoDocksHandler.getCheatSheetPresenter().ifPresent(
+                        presenter -> {
+                            if (!presenter.isCurrentlyShow(scenarioSimulationEditorPresenter.getPath())) {
+                                scenarioSimulationEditorPresenter.setCheatSheet(presenter);
+                                presenter.setCurrentPath(scenarioSimulationEditorPresenter.getPath());
+                            }
+                        });
+                break;
+            default:
+                throw new IllegalArgumentException("Invalid identifier");
+        }
+    }
```

Comment:
@danielezonca I agree to remove the duplication, as your suggestion fit wells. 
Beside that, I think the current architecture doesn't fit in the best way the logic we have. Probably a refactory of this `Wrapper`s and main presenter class should be nice, because there are some points I don't like at all:
- Logic are mixed between 'Presenter` and `Wrapper`s
- There is this circular dependency (presenter <--> wrapper) as a consequence, and it should be avoided. 
Moving part of the logic in the `dockHandler` improves a little bit the situation, and for this PR I think it's enough.

Reply:
Sure, please open a ticket for that :)